### PR TITLE
feat: process multiple directories

### PR DESCRIPTION
**Pull request by juev**
This pull request introduces the ability to run linter checks on multiple directories simultaneously, refactors the linting logic to support this feature, and adds corresponding tests and test data. Key changes include updates to the CLI, internal logic, and the addition of test cases.

### New Feature: Multi-directory Linting
* Added a new CLI command to run linter checks on multiple directories at once, with results displayed individually for each directory. (`README.md`, [README.mdR32-R39](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R32-R39))
* Introduced the `runLintMultiple` function to process multiple directories, handling errors gracefully and continuing execution for other directories. (`cmd/dmt/root.go`, [cmd/dmt/root.goL90-R121](diffhunk://#diff-75db30f73a4f406e5b7ace30c60425086735da148267773f8f7f53d329c5f897L90-R121))

### Refactoring and Code Improvements
* Refactored the `runLint` logic to remove redundant directory logging and integrate with the new `runLintMultiple` function. (`cmd/dmt/main.go`, [cmd/dmt/main.goL69](diffhunk://#diff-f19f8658998259d3e175c358cabbe50aac7df7ccf37d5ca34ffccf8181b052c0L69))
* Refactored the `Manager` initialization to separate logic into a new `initManager` method, improving modularity and readability. (`internal/manager/manager.go`, [[1]](diffhunk://#diff-794b8f34118a4167ccaababa78e3300cd3e72d5fcbc31443981337d83e1b4765R82-R85) [[2]](diffhunk://#diff-794b8f34118a4167ccaababa78e3300cd3e72d5fcbc31443981337d83e1b4765L115-R119)

### Testing and Test Data
* Added unit tests for the `runLintMultiple` function and basic tests for `lintCmdFunc` to ensure correct handling of multiple directories and edge cases. (`cmd/dmt/root_test.go`, [cmd/dmt/root_test.goR1-R27](diffhunk://#diff-d6e5da704f24435a6575e806025b439e6c8dbdb6675efef31d9055565e638cfbR1-R27))
* Added test data for multi-directory linting scenarios to validate the new functionality. (`testdata/multi-dirs-test/dir1/module.yaml`, [[1]](diffhunk://#diff-1229cebe3e7b93d3aa3f8a51519aab715abc30d7de779d7ff2726c2bf1ac4bc0R1-R2); `testdata/multi-dirs-test/dir2/module.yaml`, [[2]](diffhunk://#diff-c5e9823d04f7d608b244b96188f66956660f5fb8cbc94661d29ecf24c3ec6cd8R1-R2)

**Example**

```plain
❯ go run ./cmd/dmt lint /deckhouse/modules /deckhouse/ee/modules
time=2025-07-10T09:30:20.937+03:00 level=INFO msg="Processing directory: /deckhouse/modules"
time=2025-07-10T09:30:20.937+03:00 level=INFO msg="DMT version: devel"
time=2025-07-10T09:30:21.008+03:00 level=INFO msg="Using global values from `/deckhouse` directory"
time=2025-07-10T09:30:21.636+03:00 level=WARN msg="[WARNING] Fail: `certManager.cloudDNSServiceAccount` spec doesn't have project_id key."
time=2025-07-10T09:30:22.477+03:00 level=INFO msg="Found 49 modules"
...
🐒 [definition-file (#module)]
     Message:      Module `descriptions.en` field is required
     Module:       user-authn
     FilePath:     module.yaml


time=2025-07-10T09:30:22.991+03:00 level=INFO msg="Processing directory: /deckhouse/ee/modules"
time=2025-07-10T09:30:22.991+03:00 level=INFO msg="DMT version: devel"
time=2025-07-10T09:30:23.002+03:00 level=INFO msg="Using global values from `/deckhouse` directory"
time=2025-07-10T09:30:23.116+03:00 level=INFO msg="Found 5 modules"
time=2025-07-10T09:30:23.116+03:00 level=INFO msg="Run linters for `runtime-audit-engine` module"
...
🐒 [definition-file (#module)]
     Message:      Module `descriptions.en` field is required
     Module:       service-with-healthchecks
     FilePath:     module.yaml


time=2025-07-10T09:30:23.168+03:00 level=ERROR msg="Error processing directory /deckhouse/ee/modules: critical errors found"
exit status 1
```

### CODE REVIEW

File: cmd/dmt/root_test.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	// Test with empty directories list
+	err := runLintMultiple([]string{})
+	require.NoError(t, err, "Should handle empty directories list")
+
+	// Test with single directory
+	err = runLintMultiple([]string{"."})
+	require.NoError(t, err, "Should handle single directory")
+
+	// Test with multiple directories
+	err = runLintMultiple([]string{".", "."})
```
Comment (ldmonster) [inline cmd/dmt/root_test.go:19]: looks like we don't know which directories will process in this run
if we want to test this correctly - we must to check modules list in directories and maybe some errors.

**Review by ldmonster — APPROVED** [review]
